contrib/endpoints/src/api_manager/service_control/info.h

```diff
@@ -85,6 +86,73 @@ struct CheckResponseInfo {
   CheckResponseInfo() : is_api_key_valid(true), service_is_activated(true) {}
 };
 
+enum MetricValueType {
+  INT64,
+  BOOL,
+  DOUBLE,
+  STRING
+};
+
+struct MetricValueInfo {
+  std::string metric_name;
+  std::unordered_map<std::string, std::string> labels;
+
+  std::chrono::system_clock::time_point start_time;
+  std::chrono::system_clock::time_point end_time;
+
+  int64_t int64_value;
+  bool bool_value;
+  double double_value;
+  std::string string_value;
```

Comment:
We don't need such complicated MetricValueInfo.  All we need is:
string name;
int64_t  cost;